hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -1942,6 +1977,65 @@ public void run() {
         }
     }
 
+    /**
+     * Processes demote requests, either for this node or for other members of the cluster. Checks if any member is
+     * still in the partition table and triggers the control task.
+     * Invoked on the master node. Acquires partition service lock.
+     */
+    private class ProcessDemoteRequestsTask implements MigrationRunnable {
+        @Override
+        public void run() {
+            if (!partitionService.isLocalMemberMaster()) {
+                return;
+            }
+            partitionServiceLock.lock();
+            try {
+                final int demoteRequestCount = demoteRequestedMembers.size();
+                if (demoteRequestCount > 0) {
+                    boolean present = false;
+                    List<Member> demotedMembers = new ArrayList<>();
```

Comment:
Nit-pick: We can initialize the `ArrayList` with the size of `demoteRequestCount`

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/24617/commits/c50accbff226306616ef05fbbeba58081b91443d